_ingest-pipelines/processors/join.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: default
+title: Join
+parent: Ingest processors
+nav_order: 160
+---
+
+# Join processor
+
+The `join` processor is used to <explain what is used to do>.
+
+The following is the syntax for the `join` processor:
+
+```json
+<insert syntax example>
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `join` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+<insert the parameters>
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline, named <name>, that uses the `join` processor to <do what?>: 
+
+```json
+<insert pipeline code example>
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+<insert code example>
+```
```

Comment:
@gaobinlong Could you help me troubleshoot this pipleline example? I'm getting the following error: {
  "docs": [
    {
      "error": {
        "root_cause": [
          {
            "type": "illegal_argument_exception",
            "reason": "field [message] of type [java.lang.String] cannot be cast to [java.util.List]"
          }
        ],
        "type": "illegal_argument_exception",
        "reason": "field [message] of type [java.lang.String] cannot be cast to [java.util.List]"
      }
    }
  ]
}